### PR TITLE
Fix issue with Squeeze with empty squeeze_dims

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Resolve empty lists from tf models*

### Tickets:
 - *89090*

### CODE REVIEW

File: src/frontends/tensorflow/src/op/squeeze.cpp [hunk 1]
```diff
@@ -15,7 +15,7 @@ namespace op {
 
 OutputVector translate_squeeze_op(const NodeContext& node) {
     auto input = node.get_input(0);
-    auto axes = node.get_attribute<std::vector<int64_t>>("squeeze_dims");
+    auto axes = node.get_attribute<std::vector<int64_t>>("squeeze_dims", {});
```
Comment (rkazants) [inline src/frontends/tensorflow/src/op/squeeze.cpp:18]: Where does this attribute come from? I have not found it here: https://www.tensorflow.org/api_docs/python/tf/raw_ops/Squeeze

File: src/frontends/tensorflow/include/openvino/frontend/tensorflow/decoder.hpp [hunk 2]
```diff
@@ -11,6 +11,8 @@ namespace ov {
 namespace frontend {
 namespace tensorflow {
 
+class TENSORFLOW_API EmptyList {};
```
Comment (rkazants) [inline src/frontends/tensorflow/include/openvino/frontend/tensorflow/decoder.hpp:14]: Can we put this inside `DecoderBase` class or hide it anywhere else? Not sure it should be on the same level as `DecoderBase`

Comment (itikhono) [reply src/frontends/tensorflow/include/openvino/frontend/tensorflow/decoder.hpp:14]: I wouldn't recommend to use nested class here (inside DecoderBase I mean).
Why is it TENSORFLOW_API?

File: src/frontends/tensorflow/src/op/squeeze.cpp (hunk 1 above)
Comment (itikhono) [reply src/frontends/tensorflow/src/op/squeeze.cpp:18]: https://www.tensorflow.org/api_docs/python/tf/compat/v1/squeeze
Looks like we also need to check axis attribute, squeeze_dims has been deprecated
image

File: src/frontends/tensorflow/include/openvino/frontend/tensorflow/decoder.hpp (hunk 2 above)
Comment (mvafin) [reply src/frontends/tensorflow/include/openvino/frontend/tensorflow/decoder.hpp:14]: OVTF would use it, I guess they would need API header

Comment (itikhono) [reply src/frontends/tensorflow/include/openvino/frontend/tensorflow/decoder.hpp:14]: maybe we can move it to separate file, e.g. special_types.hpp

**Review by itikhono — APPROVED** [review]
LGTM

**Review by rkazants — APPROVED** [review]

**Pull request merged**